README.md

```diff
@@ -331,7 +331,10 @@ as follows:
 
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
-    home-manager.url = "github:nix-community/home-manager";
+    home-manager = {
+      url = "github:nix-community/home-manager";
+      inputs.nixpkgs.follows = "nixpkgs";
```

Comment:
this is potentially harmful, advanced users can use it but we should default to the mainstream method.

Reply:
could you elaborate on how this is potentially harmful?

Reply:
home-manager is guaranteed to work with the nixpkgs in its lock file, it has not been tested with the overriden version so there is a risk of failure no ?

Reply:
IMO users typically expect their entire configuration to be evaluated against a single instance of `nixpkgs` unless they specify otherwise, so I figured overriding the input would be a sensible thing to include here.

Reply:
I disagree, that's not how flakes are conceived (it doesn't default to a single nixpkgs) and I think users prefer something that work rather than shaving off a few MB on the disk. It's still possbile but shouldn't be the default I think.

Reply:
Right, but I think that would just introduce another source of confusion for users. If they update their nixpkgs input, they're _probably_ expecting to see that update reflected across all their configs.

Additionally, we're going to be tracking nixos-unstable, but what about the people whose configs use the stable branch? It's probably not a good idea for them to mix up stable with unstable (a bit more consequential than saving a few MB...), so they'd _need_ to explicitly override the nixpkgs input.

I suppose I can add a disclaimer instead that the user's home-manager config will use a different instances of nixpkgs than their nixos/darwin config instead, and just add a note on how to override that.

Reply:
if they track the stable version then they should use the home-manager flake from the stable branch. In any case, as long as they use the locked flakes, any combination should work (I think), it's only if they replace the locked inputs that they can have issues.

Reply:
we could mention the `follow` trick in a "Flake" page over the wiki https://github.com/nix-community/home-manager/wiki

Reply:
Should we then set up periodic updates for the stable branch as well to track `nixos-20.09` (and future versions)?

Reply:
That would probably be a good idea but I think we could merge already as is. I would actually merge if I could setup the FLAKEBOT_TOKEN. What are the restrictions to set it ?

Reply:
It just needs the repo permission to have push access to its flake-update branch.

Reply:
If you have write access to the repo I think you should be able to create secrets. Just note that if you create a personal access token, the flake update PRs will be opened under your name, like in [this PR](https://github.com/i077/system/pull/38).

Reply:
Note that such a token will give access to all the repos for which the user have access, as there is no way to create scoped personal access tokens. This is discussed here with a few possible solutions:
https://github.com/peter-evans/create-pull-request/issues/48

As I understand it:
- create a new user for the token and scope it down in that way
- Create a custom SSH key (which can configure access per repo level)
- Create the PR from a fork 

The reason the token is needed at all is to workaround workflows not being triggered when PR:s are opened by the default GH Actions user.

Reply:
Another solution might be to just have an organization-level secret token for this? This would also allow the same workflow to be used in other flake projects.

Reply:
@rycee thoughts on having on this automated flake.lock dependencies update?

Reply:
Sorry, I don't have much sensible input to give since I don't really know anything about flakes and even less about whether automatic lock file updates are a good idea 🙁 So I'll have to leave it up to you, let me know if there is anything I need to do with secrets and such.

If I do have to say something then I guess it would be best to have a dedicated bot account as the commit and PR author. I.e., something along the lines of @terlar's suggestion.

In the longer run, however, wouldn't the best be to have something like dependabot, which acts as a repository app? Perhaps there have already been work done in that area? Especially since I assume something like that would also be useful for the wider Nix community.

Reply:
> In the longer run, however, wouldn't the best be to have something like dependabot, which acts as a repository app? Perhaps there have already been work done in that area? Especially since I assume something like that would also be useful for the wider Nix community.

I'm working on a "flakebot" Github App to do this exact thing, but won't be able to do much on it until my semester is over in another couple weeks. For the time being, I think the workflow is still a good idea, although I still prefer an organization level flakebot token so that similar workflows could be used in other projects if necessary.

Reply:
@kclejeune Right, yeah, I had a look in the organization settings and it seem possible to create a secret that you can tie to specific repos, So that seems like a nice idea. But I guess the token should belong to the flakebot user? Who will manage that user?

@Mic92 Sorry to drag you in here but I wonder if you know of anybody else in the nix-community organization being interested in periodic CI pipeline that updates their project's `flake.lock` file? We're discussing here to create such for HM and the suggestion of @kclejeune is to add a token for a "flakebot" account as an organization level secret so that other projects in nix-community could have the possibility of adding such a pipeline.

Reply:
@rycee I created a dummy flakebot user with @i077 a while back, and we ended up deleting it because it seems unnecessary. The token belongs to `nix-community`, not to a particular user account. Whoever owns this organization can enable it in repositories where it seems fitting.

Reply:
@kclejeune, maybe I'm confused. I don't see any way to create a token for an organization. What I can see is a way to create a secret, but then I have to provide a value. Perhaps I don't have sufficient access to the nix-community organization? In the documentation I found https://docs.github.com/en/organizations/managing-organization-settings/disabling-or-limiting-github-actions-for-your-organization but it seems to be limited to `GITHUB_TOKEN`.

Reply:
> @kclejeune, maybe I'm confused. I don't see any way to create a token for an organization. What I can see is a way to create a secret, but then I have to provide a value. Perhaps I don't have sufficient access to the nix-community organization? In the documentation I found https://docs.github.com/en/organizations/managing-organization-settings/disabling-or-limiting-github-actions-for-your-organization but it seems to be limited to `GITHUB_TOKEN`.

I think this may be restricted to org admins? I could also just be mistaken and it's just not possible at all.

I was under the impression that you could generate something analogous to a personal access token at the org level. I'll setup an organization under my own account and see if I can figure it out. Otherwise, the dummy account is probably the way to go.

Reply:
I don't think this is possible, personal access tokens are per-user only, not per-organization. I am an admin of another GitHub org and I see no such option to create a token at the org level.

Reply:
@rycee Yes. Please open a ticket at https://github.com/nix-community/infra

Reply:
can we fix this ? that's the last issue before merging I think ? @Mic92 you seem to have the necessary rights, haven't you ?